src/main.rs

```diff
@@ -200,6 +243,26 @@ async fn async_main(config: Config) {
 					.with_context(|| format!("failed to bind udp socket {}", socket_addr))
 					.unwrap_or_else(|err| panic!("{err:?}"));
 				server.register_socket(udp_socket);
+			},
+			DownstreamConfig::Tls(downstream) => {
+				let cert_and_key =
+					load_cert_and_key(downstream.certificate, downstream.key)
+						.await
+						.with_context(|| "failed to load certificate or private key")
+						.unwrap_or_else(|err| panic!("{err:?}"));
+				let socket_addr = format!("{}:{}", downstream.listen, downstream.port);
+				let tcp_listener = TcpListener::bind(&socket_addr)
+					.await
+					.with_context(|| format!("failed to bind tcp socket {}", socket_addr))
+					.unwrap_or_else(|err| panic!("{err:?}"));
```

Comment:
Why are you manually implementing `expect`?

Reply:
expect does not `format!`

https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call

Reply:
Sorry, try using `.context().unwrap()` instead

Reply:
I do not like `unwrap()` this does create an ugly error message like:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value:....
```